nat/src/stateful/mod.rs

```diff
@@ -593,8 +589,8 @@ impl StatefulNat {
             debug!("{}: Found session, translating packet", self.name());
             return Self::stateful_translate(self.name(), packet, &state).and(Ok(true));
         }
-
-        let flow_key = Self::extract_flow_key(packet).ok_or(StatefulNatError::TupleParseError)?;
+        let flow_key =
+            FlowKey::try_from(Uni(&*packet)).map_err(|_| StatefulNatError::TupleParseError)?;
```

Comment:
Does `translate_packet()` really need to know how to extract a flow key, with the `Uni()` wrapper and all? What's the matter with hiding it in a function?